Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -0,0 +1,455 @@
+        //>>includeStart('debug', pragmas.debug);
+        if (gltfByteLength === 0) {
+            throw new DeveloperError('glTF byte length is zero, i3dm must have a glTF to instance.');
+        }
+        //>>includeEnd('debug');
+        var gltfView = new Uint8Array(arrayBuffer, byteOffset, gltfByteLength);
+
+        var pickObject = {
+            content : content,
+            primitive : tileset
+        };
+
+        // PERFORMANCE_IDEA: patch the shader on demand, e.g., the first time show/color changes.
+        // The pick shader still needs to be patched.
+        content._model = new Model({
+            gltf : gltfView,
+            cull : false,           // The model is already culled by 3D Tiles
+            releaseGltfJson : true, // Models are unique and will not benefit from caching so save memory
+            basePath : basePath,
+            requestType : RequestType.TILES3D,
+            modelMatrix : tile.computedTransform,
+            upAxis : tileset._gltfUpAxis,
+            shadows: tileset.shadows,
+            debugWireframe: tileset.debugWireframe,
+            incrementallyLoadTextures : false,
+            vertexShaderLoaded : getVertexShaderCallback(content),
+            fragmentShaderLoaded : getFragmentShaderCallback(content),
+            uniformMapLoaded : batchTable.getUniformMapCallback(),
+            pickVertexShaderLoaded : getPickVertexShaderCallback(content),
+            pickFragmentShaderLoaded : batchTable.getPickFragmentShaderCallback(),
+            pickUniformMapLoaded : batchTable.getPickUniformMapCallback(),
+            addBatchIdToGeneratedShaders : (batchLength > 0), // If the batch table has values in it, generated shaders will need a batchId attribute
+            pickObject : pickObject
+        });
+    }
+
+    function createFeatures(content) {
+        var tileset = content._tileset;
+        var featuresLength = content.featuresLength;
+        if (!defined(content._features) && (featuresLength > 0)) {
+            var features = new Array(featuresLength);
+            for (var i = 0; i < featuresLength; ++i) {
+                features[i] = new Cesium3DTileFeature(tileset, content, i);
+            }
+            content._features = features;
+        }
+    }
+
+    /**
+     * Part of the {@link Cesium3DTileContent} interface.
+     */
+    Batched3DModel3DTileContent.prototype.hasProperty = function(batchId, name) {
+        return this._batchTable.hasProperty(batchId, name);
+    };
+
+    /**
+     * Part of the {@link Cesium3DTileContent} interface.
+     */
+    Batched3DModel3DTileContent.prototype.getFeature = function(batchId) {
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId >= featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + (featuresLength - 1) + ').');
```

Comment:
I'll stop mentioning it, please review all new errors and if the error comes from data instead of code, they should be runtime errors.